kumascript/src/live-sample.js

```diff
@@ -68,100 +62,37 @@ const LIVE_SAMPLE_HTML = `
-}
-
-class LiveSampleID {
-  constructor(id, source, token) {
-    this.id = id;
-    this.source = source;
-    this.token = token;
-  }
-
-  createFlaw(message) {
-    return new MacroLiveSampleError(
-      message instanceof Error ? message : new Error(message),
-      this.source,
-      this.token
-    );
-  }
-}
-
-function getLiveSampleIDs(slug, source) {
-  // Given a slug and its raw source HTML, parses the source and returns
-  // a list of live-sample ID's to be extracted from this document.
-  const tokens = Parser.parse(source);
-  const currentSlug = normalizeSlug(slug);
-  // Loop through the tokens, looking for calls to the EmbedLiveSample macro.
-  // The first argument to the call is the live-sample ID, and there may also
-  // be an optional fifth argument that specifies a slug, that may or may not
-  // be different from the current slug, from which to extract the sample ID.
-  const result = [];
-  for (const token of tokens) {
-    if (token.type !== "MACRO") continue;
-    const normalizedMacroName = normalizeMacroName(token.name);
-    if (normalizedMacroName === "inheritancediagram") {
-      // The InheritanceDiagram is a special macro that, unlike other EmbedLiveSample
-      // macros in that it itself renders the "EmbedLiveSample" with a set of
-      // hardcoded options. The only thing that is variable and comes from the
-      // the raw HTML is the size. So make an exception for this otherwise, the
-      // HTML element ID is lost and this function won't discovered that it is
-      // in fact a live sample here.
-      result.push(new LiveSampleID("inheritance_diagram", source, token));
-    } else if (
-      (normalizedMacroName === "embedlivesample" ||
-        normalizedMacroName === "livesamplelink") &&
-      token.args.length
-    ) {
-      // Some of the localized pages URI-encode their first argument,
-      // the live-sample ID, even though they don't need to do that,
-      // so let's first call "safeDecodeURIComponent" just in case.
-      const sampleID = slugify(safeDecodeURIComponent(token.args[0]));
-      const sampleIDObject = new LiveSampleID(sampleID, source, token);
-      if (token.args.length > 4) {
-        // Some calls to EmbedLiveSample explicitly specify a slug from which
-        // to extract the live sample ID in the 5th argument (4th index), so
-        // we can't just assume that all of the sample ID's are to be extracted
-        // from the current document.
-        const slugArg = normalizeSlug(token.args[4]);
-        if (slugArg && slugArg !== currentSlug) {
-          // If this live-sample is to be extracted from another document,
-          // don't return it as part of this document's list of sample ID's.
-          continue;
+  // and return the HTML of the live-sample pages for the given
+  // document or else collect flaws
+  const $ = cheerio.load(renderedHTML);
+  return $("iframe")
+    .filter((i, iframe) => $(iframe).attr("src").includes("/_sample_."))
```

Comment:
@Gregoor I discovered the cause the build failure. It's because this line is too inclusive. We should only build a live-sample page for live-sample `iframe`s whose `src` is for the current page, so the fix simply becomes:
```suggestion
    .filter((i, iframe) => $(iframe).attr("src").includes(`${uri}/_sample_.`))
```

Reply:
@Gregoor I'll commit my suggestion (which I've already tried locally), so we can see what else we run into with the tests.

Reply:
Ah! I did not consider that. Great find, thank you!